src/components/TokenExpired.js

```diff
@@ -1,25 +1,91 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 import { connect } from 'react-redux'
+import { SessionTimeout } from 'patternfly-react'
 
 import ErrorAlert from './ErrorAlert'
 import { msg } from '_/intl'
 
+import { logout } from '_/actions'
+
+const MAX_INACTIVE_TIME = 60 * 20 // 20 minutes
+const TIME_TO_DISPLAY_MODAL = 30 // 30 seconds
+
 // TODO: allow the user to cancel the automatic reload?
 // If so, change config.isTokenExpired to false and add additional check to doCheckTokenExpired() before actual reload
-const TokenExpired = ({ config }) => {
-  if (!config.get('isTokenExpired')) {
-    return null
+class TokenExpired extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      expired: false,
+      counter: MAX_INACTIVE_TIME,
+    }
+
+    this.dropCounter = this.dropCounter.bind(this)
+    this.decrementCounter = this.decrementCounter.bind(this)
+
+    document.body.addEventListener('mousemove', this.dropCounter)
+    this.timer = setInterval(this.decrementCounter, 1000)
```

Comment:
Shouldn't the component really mount before the timeout starts?

Reply:
Actually no, because mount cycle in React has `constructor` and `componentDidMount`, timer is not depend on inner content of component, so it doesn't really matter where to have it.

Reply:
As long as the component won't be constructed and sit around for awhile before getting mounted (for example while fetching the initial login data).

I mean even then it shouldn't matter much.  Even if the initial load takes 5 minutes and the inactivity timer is down from 30 minutes to 25, the user is probably going to start clicking once it opens.  Well that assumes a user will wait around for 5 minutes...